### DISCUSSION TITLE
Reverse expand on a many-to-many relation

### DISCUSSION THREAD

**Discussion by FarazPatankar** [pocketbase/pocketbase#4149] [Q&A]
Hey, so I have two tables: `touristVisas` and `visaDetails`. `visaDetails` can often be the same for multiple `touristVisas`. So, I have a relation field on the `visaDetails` table called `visa` which points to multiple `touristVisas` objects.

Now, when I am viewing a `touristVisas` object, I'd like to expand and fetch the `visaDetails` for it. Does Pocketbase support this or will I manually have to add a relation to every `touristVisas` object?

Alternatively, do you have any other suggestions on how to do this?

**Comment by FarazPatankar** [comment]
I don't believe this is possible so I ended up creating a `View Collection` with the following SQL:
```
SELECT
  tv.id,
  -- other fields go here

  vd.slug as visaDetailsSlug
FROM
  touristVisas tv
LEFT JOIN 
  visaDetails vd ON tv.id IN (
        SELECT json_each.value
        FROM json_each(vd.visa)
    );
```

I'd still love to know if the original query is doable or if what I've done is optimal as my SQL chops are very basic, haha!

**Reply by jonshipman** [reply]
Views are a very powerful tool (still trying to break my mysql habit of coding SQL solutions for every problem). So far, I've been liking having custom views for specific screens when needed. Like, a listing table and a detailed view don't need the same fields available. I'll create a `jobs_list` and `jobs_view` view tables and select only the fields used on their respective screens. Only if I run into a problem using the base tables as it's on a case-by-case basis.

If you need a more generalized answer - determine which table is the primary table and have the child reference its id. Ask yourself, "does touristVistas belong to vistaDetails or does vistaDetails belong to touristVistas?" Having a linking table like you did is also good, though I'd probably set it up mulitple-to-multiple as in `id`, `vistaDetail` id, and `touristVista` id and not bother with the multiple relations instead opting to do a `.getFullList` on the client app.
